src/Node.cpp

```diff
@@ -177,9 +177,6 @@ propagateData(std::shared_ptr<NodeData> nodeData,
 {
   _nodeDataModel->setInData(nodeData, inPortIndex);
 
-  _nodeGeometry.recalculateSize();
-  _nodeGraphicsObject->setGeometryChanged();
-  _nodeGraphicsObject->update();
```

Comment:
I think this is misplaced because think about this scenario:
- Node is deleted,
- The NodeDataModel is deleted
- An empty shared ptr is passed to propagateData
- recalculateSize needs the NodeDataModel, so segfault!

(This happened to me)